app/tests/unit/pages/project-revision/[projectRevision]/index.test.tsx

```diff
@@ -419,4 +434,87 @@ describe("The Create Project page", () => {
     pageTestingHelper.renderPage();
     expect(screen.getByText("Submit")).toBeEnabled();
   });
+
+  it("routes to the project list page when discarding a project revision and isFirstRevision is true", () => {
+    mocked(useRouter).mockReturnValue({
+      replace: jest.fn(),
+      push: jest.fn(),
+    } as any);
+
+    const mockresolver = {
+      ProjectRevision() {
+        return {
+          id: "mock-proj-rev-id",
+          isFirstRevision: true,
+          projectByProjectId: null,
+          projectFormChange: {
+            id: "mock-project-form-id",
+            newFormData: {
+              someProjectData: "test2",
+            },
+          },
+        };
+      },
+    };
+    pageTestingHelper.loadQuery(mockresolver);
+    pageTestingHelper.renderPage();
+
+    userEvent.click(screen.queryByText("Discard Project Revision"));
+    userEvent.click(screen.queryByText("Proceed"));
+
+    act(() => {
+      pageTestingHelper.environment.mock.resolveMostRecentOperation({
+        data: { input: "asdf" },
+      });
+    });
+
+    expect(useRouter().push).toHaveBeenCalledWith({
+      pathname: "/cif/projects/",
+    });
+  });
+
+  it("routes to the project overview when discarding project revision and isFirstRevision is false", () => {
+    mocked(useRouter).mockReturnValue({
+      replace: jest.fn(),
+      push: jest.fn(),
+    } as any);
+
+    const mockresolver = {
+      ProjectRevision() {
+        return {
+          id: "mock-proj-rev-id",
+          isFirstRevision: false,
+          projectByProjectId: {
+            latestCommittedProjectRevision: {
+              id: "last-revision-id",
+            },
+          },
+          projectFormChange: {
```

Comment:
`projectFormChange` can go from here too, everything above it is needed in this one though.